compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -262,6 +262,19 @@ object Contexts {
     /** AbstractFile with given path, memoized */
     def getFile(name: String): AbstractFile = getFile(name.toTermName)
 
+    def getSiblingClassfile(tastyFile: AbstractFile): AbstractFile =
+      base.siblingClassfiles.getOrElseUpdate(tastyFile, {
```

Comment:
we could possibly check before caching to see if Zinc is enabled.

Reply:
Did you check how many cache hits we get?